File: ELEC881_Tabnine/src/No Description Prompt/Java script (JS)/q7-super-ugly-number.js
/**
 * @param {number} n
 * @param {number[]} primes
 * @return {number}
 */
var nthSuperUglyNumber = function(n, primes) {
    const ugly = [1];
    const indices = primes.map(() => 0);
    while (ugly.length < n) {
        const nextUgly = Math.min(...primes.map(prime => prime * ugly[indices[primes.indexOf(prime)]]));
        ugly.push(nextUgly);
        indices[primes.indexOf(nextUgly)]++;
    }
    return ugly[n - 1];
};

File: ELEC881_Tabnine/src/No Description Prompt/Python/q7-super-ugly-number.py
class Solution(object):
    def nthSuperUglyNumber(self, n, primes):
        """
        :type n: int
        :type primes: List[int]
        :rtype: int
        """
        if n <= 0:
            return 0

        ugly = [0] * n
        ugly[0] = 1
        index = [0] * len(primes)

        for i in range(1, n):
            next_ugly = float('inf')
            for j in range(len(primes)):
                next_ugly = min(next_ugly, ugly[index[j]] * primes[j])
            ugly[i] = next_ugly

            for j in range(len(primes)):
                if ugly[i] == ugly[index[j]] * primes[j]:
                    index[j] += 1

        return ugly[n - 1]







        